### ISSUE TITLE
firefox allocation size overflow editor publishing 1,5m+ polygons

### ISSUE THREAD

**Issue by ThomasPahler** [mrdoob/three.js#9187]
##### Description of the problem 

allocation size overflow when publishing a scene frome editor with over 1,5 million polygons, 1,1million still works fine

allocation size overflow
Menubar.File/<()
 Menubar.File.js:221
bound ()
 self-hosted:752
 Menubar.File.js:221:12

##### Three.js version

- [ ] Dev
- [x] r77
- [ ] ...

##### Browser

- [ ] All of them
- [ ] Chrome
- [x] Firefox
- [ ] Internet Explorer

##### OS

- [x] All of them
- [ ] Windows
- [ ] Linux
- [ ] Android
- [ ] IOS

**Comment by mrdoob** [comment]
Yeah... Working on that...

**Comment by ThomasPahler** [comment]
here (http://www.unmht.org/forum/en/5108.html) someone said that "Firefox's JavaScript engine can handle up to 256M chars per each string", might be the problem here, so it had to be splitted or reduced in some way.
i noticed that the scene export still works, maybe it just has a higher limit?

**Comment by mrdumb** [comment]
> allocation size overflow when publishing a scene frome editor with over 1,5 million polygons, 1,1million still works fine

Anything above 1 million poligons in a browser is a bad idea from many points of view.
You need to learn how to optimize your stuff.
Lower poly and "smart" textures is the key to success.

**Comment by makc** [comment]
> i noticed that the scene export still works, maybe it just has a higher limit?

that's because of 35e5c7

**Comment by MatejJan** [comment]
> Anything above 1 million polygons in a browser is a bad idea from many points of view.
You need to learn how to optimize your stuff.

That's not an answer for everyone. I'm using THREE.js for scientific visualization in a videowall center with multiple Quadro graphics cards that can easily handle huge polygon counts. It shouldn't be show-stopper then with loading.

**Comment by zhanghaidi** [comment]
Yes, this issue has nothing to do with thee.js rendering. It seems to be related to the memory limit of the jsV8 engine. Can I choose a scheme for batch export based on objects?

**Comment by makc** [comment]
perhaps something like [this](https://github.com/Faleij/json-stream-stringify#complete-example) could help here, but then you would need your own replacement of fflate.strToU8 to make a normal array (that you could grow) instead
